### PR TITLE
Bumps library versions to latest

### PR DESCRIPTION
**Pull request by adriancole**
NOTE: mongo in a patch suddenly started leaking [endSessions](https://docs.mongodb.com/manual/reference/command/endSessions/) admin commands into the listener. I've filtered admin commands out similar to how we do in p6spy and how we choose intentionally to not trace admin commands in kafka

https://github.com/mongodb/mongo-java-driver/commit/e0572db771b682f6c50006965f64c5de1f17a1b3

### CODE REVIEW

File: brave-tests/src/main/java/brave/test/propagation/PropagationTest.java [hunk 1]
```diff
@@ -180,8 +180,9 @@ public MapEntry() {
       Map<Object, String> map = new LinkedHashMap<>();
       injector.inject(ctx, map);
 
-      assertThat(extractor.extract(map).context())
```
Comment (adriancole) [inline brave-tests/src/main/java/brave/test/propagation/PropagationTest.java:183]: assertj got weird and leaked data which I didn't feel like hunting down. This is a classloader test

File: brave/src/test/java/brave/handler/MutableSpanTest.java [hunk 2]
```diff
@@ -578,7 +579,7 @@
         MutableSpan other = otherConstructor.get();
         assertThat(span).isNotSameAs(other); // sanity
         assertNeitherEqualNorShareHashCode(span, other);
-        assertThat(span).usingRecursiveComparison().isNotEqualTo(other); // double check our impl
+        assertThat(Objects.deepEquals(span, other)).isFalse(); // double check our impl
```
Comment (adriancole) [inline brave/src/test/java/brave/handler/MutableSpanTest.java:582]: assertj's recursive can't tell the difference between two exceptions with the same value state anymore

File: instrumentation/kafka-clients/src/main/java/brave/kafka/clients/TracingConsumer.java [hunk 3]
```diff
@@ -313,6 +313,11 @@ public ConsumerGroupMetadata groupMetadata() {
     return delegate.groupMetadata();
   }
 
+  // Do not use @Override annotation to avoid compatibility issue version < 2.6
+  public void enforceRebalance() {
+    delegate.enforceRebalance();
```
Comment (adriancole) [inline instrumentation/kafka-clients/src/main/java/brave/kafka/clients/TracingConsumer.java:318]: typical maintenance and without this we can't really trace kafka 2.6 as it won't completely implement the interface

File: instrumentation/mongodb/src/main/java/brave/mongodb/TraceMongoCommandListener.java [hunk 4]
```diff
@@ -61,11 +61,13 @@
    * all callbacks happen on the same thread.
    */
   @Override public void commandStarted(CommandStartedEvent event) {
+    String databaseName = event.getDatabaseName();
+    if ("admin".equals(databaseName)) return; // don't trace commands like "endSessions"
```
Comment (adriancole) [inline instrumentation/mongodb/src/main/java/brave/mongodb/TraceMongoCommandListener.java:65]: super strange that in a patch "admin" database commands suddenly show up. Due to order of things, we would only see "endSessions"! This explicitly gets rid of this similar to p6spy until/unless someone asks for full admin coverage and in that case end w/o start probably is undesirable.

File: instrumentation/spring-rabbit/src/test/java/brave/spring/rabbit/ITSpringRabbit.java [hunk 5]
```diff
@@ -61,14 +67,41 @@
...
   static final Exchange exchange_batch = topicExchange("test-exchange-1").durable(true).build();
   static final Queue queue_batch = new Queue(TEST_QUEUE_BATCH);
-  static final Binding binding_batch = bind(queue_batch).to(exchange_batch).with("test.binding.1").noargs();
+  static final Binding binding_batch =
+    bind(queue_batch).to(exchange_batch).with("test.binding.1").noargs();
+
+  static final String IMAGE = "rabbitmq:3.8-management-alpine";
+  static final int RABBIT_PORT = 5672;
+
+  static final class RabbitMQContainer extends GenericContainer<RabbitMQContainer> {
```
Comment (adriancole) [inline instrumentation/spring-rabbit/src/test/java/brave/spring/rabbit/ITSpringRabbit.java:76]: same as our zipkin image, inlined and also lets us skip docker if we want.

File: pom.xml [hunk 6]
```diff
@@ -82,55 +83,58 @@
...
     <powermock.version>2.0.7</powermock.version>
-    <mockito.version>3.3.3</mockito.version>
+    <mockito.version>3.5.10</mockito.version>
     <jersey.version>2.31</jersey.version>
-    <jmh.version>1.23</jmh.version>
+    <!-- must align with kafka version https://github.com/charithe/kafka-junit -->
+    <kafka-junit.version>4.1.10</kafka-junit.version>
     <testcontainers.version>1.14.3</testcontainers.version>
 
+    <animal-sniffer-maven-plugin.version>1.19</animal-sniffer-maven-plugin.version>
```
Comment (adriancole) [inline pom.xml:122]: ordered lexicographical as tired of scanning

**Review by adriancole — COMMENTED** [review]
notes!

File: pom.xml (hunk 6 above)
Comment (jeqo) [reply pom.xml:122]: thanks!! 🙏🏽

**Review by jeqo — APPROVED** [review]
LGTM, thanks @adriancole !

**Review by anuraaga — APPROVED** [review]

**Comment by adriancole** [comment]
thanks for the look @jeqo and @anuraaga !

**Pull request merged**